src/main/java/io/confluent/idesidecar/restapi/cache/ClientConfigurator.java

```diff
@@ -0,0 +1,246 @@
+    return getKafkaClientConfig(connection, cluster, null, redact, adminClientSidecarConfigs);
+  }
+
+  /**
+   * Get the KafkaConsumer configuration for connection and Kafka cluster with the specified IDs.
+   * This method looks up the {@link ConnectionState} and {@link KafkaCluster} objects,
+   * and will throw exceptions if the connection or cluster does not exist.
+   *
+   * <p> This can optionally redact sensitive values in properties, such as if generating a
+   * sample configuration for display or logging.
+   *
+   * @param connectionId          the ID of the connection
+   * @param clusterId             the ID of the Kafka cluster
+   * @param includeSchemaRegistry whether to include configuration properties for Schema Registry
+   * @param redact                whether to redact sensitive properties
+   * @return the AdminClient configuration properties
+   * @throws ConnectionNotFoundException if the connection does not exist
+   * @throws ClusterNotFoundException    if the cluster does not exist
+   * @see #getKafkaClientConfig(ConnectionState, KafkaCluster, SchemaRegistry, boolean, Map)
+   */
+  public Map<String, Object> getConsumerClientConfig(
+      String connectionId,
+      String clusterId,
+      boolean includeSchemaRegistry,
+      boolean redact
+  ) throws ConnectionNotFoundException, ClusterNotFoundException {
+    // Find the connection and cluster, or fail if either does not exist
+    var connection = connections.getConnectionState(connectionId);
+    var cluster = clusterCache.getKafkaCluster(connectionId, clusterId);
+
+    // Maybe look up the SR for the Kafka cluster
+    SchemaRegistry sr = null;
+    if (includeSchemaRegistry) {
+      sr = clusterCache.getSchemaRegistryForKafkaCluster(connectionId, cluster);
+    }
+    // Return the Consumer config
+    return getKafkaClientConfig(connection, cluster, sr, redact, Map.of());
+  }
+
+  /**
+   * Get the KafkaProducer configuration for connection and Kafka cluster with the specified IDs.
+   * This method looks up the {@link ConnectionState} and {@link KafkaCluster} objects,
+   * and will throw exceptions if the connection or cluster does not exist.
+   *
+   * <p> This can optionally redact sensitive values in properties, such as if generating a
+   * sample configuration for display or logging.
+   *
+   * @param connectionId          the ID of the connection
+   * @param clusterId             the ID of the Kafka cluster
+   * @param includeSchemaRegistry whether to include configuration properties for Schema Registry
+   * @param redact                whether to redact sensitive properties
+   * @return the AdminClient configuration properties
+   * @throws ConnectionNotFoundException if the connection does not exist
+   * @throws ClusterNotFoundException    if the cluster does not exist
+   * @see #getKafkaClientConfig(ConnectionState, KafkaCluster, SchemaRegistry, boolean, Map)
+   */
+  public Map<String, Object> getProducerClientConfig(
```

Comment:
The producer config has SR client configs, but the consumer does not (since we never have the consumer deserialize). Plus, I have a change coming in a moment that adds `acks=all` for producers.

Reply:
> The producer config has SR client configs, but the consumer does not (since we never have the consumer deserialize).

`getConsumerClientConfig` seems to set the SR client, too.

> Plus, I have a change coming in a moment that adds acks=all for producers.

IIRC, `acks=all` is the default for producers.

Reply:
Okay. I've refactored the code into a common helper method. But I left `acks=all` because I think that's useful to document, even if it is the default. And if we generate configs for users to copy, having that will align with what the CCloud UI does when it generates client configs.